x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -26,22 +28,87 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
               if (ruleDataClient.isWriteEnabled() && numAlerts) {
                 const commonRuleFields = getCommonAlertFields(options);
 
+                const CHUNK_SIZE = 10000;
+                const alertChunks = chunk(alerts, CHUNK_SIZE);
+                const filteredAlerts: typeof alerts = [];
+
+                for (const alertChunk of alertChunks) {
```

Comment:
Few things here... `10000` seems like a big chunk size and could generate some really large request/response bodies. However, it seems like we aren't likely to encounter 10k alerts here unless the customer has turned `max_signals` way up, is that correct?

If we reduce the chunk size, we could potentially make this more efficient by getting a map of promises and awaiting them all at once (first for the query / building of uuid map and then for the actual writing of the alerts). Might be worth experimenting.

Also, another quick note... we use `10000` here and elsewhere as a hard-coded value, but if the customer has managed to reduce `max_result_window` from the default, these requests could potentially fail. Can we retrieve the config value from ES and use that instead?

Reply:
Yeah unless max_signals is increased significantly we won't see 10k alerts here. I thought 10k would be reasonable since we're only including the alert IDs in the request, not the full alert documents which could be arbitrarily large. So it should be at most (36 characters per UUID) * (10k alerts) ~= 360k characters in the body. The response similarly does not any retrieve full alert docs, it's just making a bucket for each unique alert UUID so its size should be capped at a couple MB at most.
 
I think the `10000` value is controlled by the cluster setting `search.max_buckets` (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-settings.html#search-settings-max-buckets)...I think the error we get from the request if a cluster is set with this value too low would be useful enough?

Reply:
Makes sense to me. 👍